pkg/provider/azure_loadbalancer.go

```diff
@@ -2411,7 +2411,14 @@ func (az *Cloud) reconcileSecurityGroup(clusterName string, service *v1.Service,
 		sourceRanges          = accessControl.SourceRanges()
 		allowedServiceTags    = accessControl.AllowedServiceTags()
 		allowedIPRanges       = accessControl.AllowedIPRanges()
-		sourceAddressPrefixes = accessControl.IPV4Sources()
+		sourceAddressPrefixes = func() []string {
+			// AKS v1.26 does not support dual stack, so it returns the single stack source ranges instead.
+			if service.Spec.IPFamilies[0] == v1.IPv4Protocol {
+				return accessControl.IPV4Sources()
+			} else {
```

Comment:
else is not needed here

Reply:
I would skip this for now as it’s just one line. But, I’m with you on reducing nested or deep branch as much as we can.